lean4-infoview/src/infoview/userWidget.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+import type { Location } from 'vscode-languageserver-protocol';
+
+import { EditorContext, RpcContext } from './contexts';
+import { Details } from './collapsing';
+import { mapRpcError, } from './rpcInterface';
+import { DocumentPosition, useAsync, useEventResult } from './util';
+import { ErrorBoundary } from './errors';
+import { RpcSessions } from './rpcSessions';
+import { isRpcError, RpcErrorCode } from '@lean4/infoview-api';
+
+export interface GetWidgetResponse {
```

Comment:
Some documentation would really be helpful here...

Reply:
working on it!